std/algebra/emulated/fields_bn254/doc.go

```diff
@@ -0,0 +1,7 @@
+// Package fields_bn254 implements the fields arithmetic of the Fp12 tower
+// used to compute the pairing over the BN254 curve.
+//
+//	𝔽p²[u] = 𝔽p/u²+1
+//	𝔽p⁶[v] = 𝔽p²/v³-9-u
+//	𝔽p¹²[w] = 𝔽p⁶/w²-v
+package fields_bn254
```

Comment:
What do you think about making this package `fields_bn254` internal? Under `std/algebra/emulated/internal/fields_bn254`. When we have type alias `GTEl = E12` in `sw_bn254`, then it should be sufficient?

My reasoning for making it internal is that then we have more freedom in reimplementing, optimising it without having to worry that there are any external users. I would still keep `sw_emulated` and `sw_bn254` non-internal.

Reply:
that can make sense for now but I suspect we would have later use-cases of `fields_bn254` as an independent package (e.g. Protocols that use GT-arithmetic like SnarkPack or SSLE would need to export e12.go and e6.go for compression).